packages/trunx/src/classNames.ts

```diff
@@ -1,15 +1,39 @@
+export type ClassNamesArg<ClassName extends string> =
+  | ClassName
+  | { [key in ClassName]: unknown }
+  | ClassNamesArg<ClassName>[]
+
+/**
+ * Utility for conditionally joining CSS classes together.
+ *
+ * @example
+ * type C = 'foo' | 'bar' // my CSS classes
+ * classNames<C>('foo', 'bar') // 'foo bar'
+ * classNames<C>('foo', ['bar']) // 'foo bar'
+ * classNames<C>({ foo: true }, { bar: false }) // 'foo'
+ */
+export const classNames = <T extends string>(...args: ClassNamesArg<T>[]): string =>
+  args
+    .map((arg) => {
+      if (typeof arg === 'string') return arg
+      if (Array.isArray(arg) && arg.length) return classNames<string>(...arg)
+      if (arg && typeof arg === 'object')
+        return classNames<string>(
+          Object.entries(arg)
```

Comment:
I would not allocate more variables then necessary since it is going to be used a lot.
I prefer add more comments to improve readability.
Done here f3956d8bf52999ead4ea0baf9d3f444c262ac628